### ISSUE TITLE
hiera 5 environmental compatible

### ISSUE THREAD

**Issue by jfgibbins** [jovandeginste/hiera-router#1]
Nice project, but any chance you might ever update it to make it compatible with hiera 5 at the environmental and module level.        hiera3_backend is only applicable at the global level.  thanks

**Comment by jovandeginste** [comment]
It is on the roadmap after the summer.

**Comment by jfgibbins** [comment]
Sounds great.  I'd love to be able to use this with consul and vault, but global just doesn't work in the environment.  I'll keep in eye on this and test whenever I can.

**Comment by jovandeginste** [comment]
We have the same requirement, although we are looking to eliminate the router first and fork the vault backend to be more flexibel.

**Comment by jfgibbins** [comment]
Really curious about what you're thinking about to improve over your setup.  I had rather liked the way router was done, if nothing else than most of my config is in plain yaml, and never liked the idea of iterating through every backend for every key, just for a couple pieces of data like from vault.  Plus, having the yaml be the first place to look, and have it tell you where it is, I thought was very well done.

**Comment by jovandeginste** [comment]
With hiera 5 and a more flexibel vault we can add a rule in the hiera.yaml, in the sense of "use this (vault) backend if the lookup key starts with 'vault_', first strip that prefix". This moves the router functionality (for us) to the hiera.

**Comment by jfgibbins** [comment]
think someone already got a start on that idea, check out https://github.com/davealden/hiera-vault

**Comment by jfgibbins** [comment]
thinking about playing with that one, it's a fork of jsok, but he hasn't done one for consul, and I'd like that as well, preferably vault and consul same design and format

**Comment by jovandeginste** [comment]
Yes that fork was what I'm thinking of, I'm on my phone and getting late here so I was lazy to look it up :-)

**Comment by jovandeginste** [comment]
@jfgibbins As an update, I migrated our Puppet setup entirely to Puppet 5 + Hiera 5 + @davealden's [hiera-vault fork](https://github.com/davealden/hiera-vault) (with a few changes, see our PR), and I'm happy to report everything now works as I want it without the router. Maybe I'm an exception, but I'm always happy to retire software when it's no longer needed (and the router was more like a hack to fix a hiatus).

**Comment by jfgibbins** [comment]
@jovandeginste awesome, working on some new firewall testing here, so wreaking havoc on my network, but I'll probably switching back over to work on puppet in a few days, look forward to giving it a try :)
